gateway/ln-gateway/src/federation_manager.rs

```diff
@@ -101,4 +102,18 @@ impl FederationManager {
             .retain(|_, fid| *fid != federation_id);
         Ok(())
     }
+
+    pub async fn get_client_for_scid(
+        &self,
+        short_channel_id: u64,
+    ) -> Option<Spanned<ClientHandleArc>> {
+        let scid_to_feds = self.scid_to_federation.read().await;
+        let clients = self.clients.read().await;
+
+        let federation_id = scid_to_feds.get(&short_channel_id)?;
+        // TODO(tvolk131): Cloning the client here could cause issues with client
+        // shutdown (see `remove_client` above). Perhaps this function should take a
+        // lambda and pass it into `client.with_sync`.
```

Comment:
Sorry, kind of tough to tell with everything moving around. Was this a problem before?

Reply:
Yes. On master, you can search for `.clients` in [`gateway/ln-gateway/src/lib.rs`](https://github.com/fedimint/fedimint/blob/master/gateway/ln-gateway/src/lib.rs) and find where `.clone()` or `.cloned()` are called on it. I'm guessing it hasn't surfaced any issues yet, but it strikes me as bad practice and could lead to the issue described in this comment. IMO ideally we should uphold the invariant that there is never more than one reference to a client by simply not using an `Arc` to hold it.

Reply:
I think we had some sort of reference counting in `ClientArc` that solved this issue. Maybe @dpc or @maan2003 remembers. I think if we don't use an `Arc` it would prevent us from doing operations in parallel on the same client (maybe that's ok), e.g lookup the balance via the REST API and initiate a payment